flink-python/src/main/java/org/apache/flink/streaming/api/operators/python/AbstractPythonFunctionOperator.java

```diff
@@ -275,4 +287,27 @@ private void invokeFinishBundle() throws Exception {
 			}
 		}
 	}
+
+	protected PythonEnvironmentManager createPythonEnvironmentManager() throws IOException {
+		PythonDependencyManager dependencyManager = PythonDependencyManager.create(
+			getExecutionConfig().getGlobalJobParameters().toMap(),
+			getRuntimeContext().getDistributedCache());
+		PythonEnv pythonEnv = getPythonEnv();
+		if (pythonEnv.getExecType() == PythonEnv.ExecType.PROCESS) {
+			String taskManagerLogFile = getContainingTask()
+				.getEnvironment()
+				.getTaskManagerInfo()
+				.getConfiguration()
+				.getString(ConfigConstants.TASK_MANAGER_LOG_PATH_KEY, System.getProperty("log.file"));
+			String logDirectory = taskManagerLogFile == null ? null : new File(taskManagerLogFile).getParent();
```

Comment:
In which case taskManagerLogFile is null?

Reply:
e.g. running IT cases in IDE.